src/sqlfluff/api/simple.py

```diff
@@ -68,13 +68,12 @@ def parse(sql, dialect="ansi"):
             to be linted. Defaults to `ansi`.
 
     Returns:
-        :obj:`FileSegment` containing the parsed structure, or None
-            on failure.
+        :obj:`ParsedString` containing the parsed structure.
     """
     sql = _unify_str_or_file(sql)
     linter = Linter(dialect=dialect)
-    parsed, violations, _ = linter.parse_string(sql)
+    parsed = linter.parse_string(sql)
     # If we encounter any parsing errors, raise the first one we find.
-    if violations:
-        raise violations[0]
+    if parsed.violations:
+        raise parsed.violations[0]

```

Comment:
Interesting! I didn't know we were only raising the first violation for parsing, I wonder if returning an Exception which encompasses all of the violations would be useful 🤔  (out of scope of the PR)

Reply:
For parsing through the command line we display all of them. It's only in the simple api which we don't. I wanted something which provided a straightforward API which was easy to use, but that for more advanced use cases, developers would start to use the main `Linter()` class directly.

I think your idea of raising an exception with all of them is a strong one! Can you make a github issue for that with what you had in mind?

Reply:
created https://github.com/sqlfluff/sqlfluff/issues/569